### PR TITLE
Eregcsc 790 goto ver hash

### PR DESCRIPTION
**Pull request by PhilR8**

### CODE REVIEW

File: regulations/static/regulations/css/scss/partials/_compare.scss [hunk 1]
```diff
@@ -1,6 +1,9 @@
 @import "../application_settings";
 
 .view-and-compare {
+    display: flex;
+    justify-content: center;
+    height: $view_compare_height;
```
Comment (PhilR8) [inline regulations/static/regulations/css/scss/partials/_compare.scss:6]: explicit height so we can predictably scroll cross-browser

File: regulations/static/regulations/css/scss/partials/_compare.scss [hunk 2]
```diff
@@ -47,18 +50,21 @@
         right: $spacer-4;
         cursor: pointer;
         height: auto;
+        border: none;
+        background: none;
+        height: 36px;
+        width: 36px;
+        display: flex;
+        align-items: center;
+        justify-content: center;
```
Comment (PhilR8) [inline regulations/static/regulations/css/scss/partials/_compare.scss:59]: give X "button" room for taps, match other buttons around site

File: regulations/static/regulations/js/go-to-version.js [hunk 3]
```diff
@@ -5,25 +5,57 @@ export function goToVersion() {
     }
 
     const options = document.querySelectorAll("#view-options [data-url]");
+    select.addEventListener("change", function () {
+        location.href =
+            this.options[this.selectedIndex].dataset.url + location.hash;
```
Comment (PhilR8) [inline regulations/static/regulations/js/go-to-version.js:10]: append current hash to option URL on change

File: regulations/static/regulations/js/go-to-version.js [hunk 4]
```diff
@@ -5,25 +5,57 @@ export function goToVersion() {
...
+    // just re-hide version select bar
+    closeBtn.addEventListener("click", (e) => {
+        if (e.currentTarget.href === location.href) {
+            const viewButton = document.querySelector("#view-button");
+            viewButton.setAttribute("data-set-state", "show");
+            viewButton.setAttribute("data-state", "not-selected");
+            const versionSelectBar = document.getElementById(
+                "view-and-compare"
+            );
+            versionSelectBar.setAttribute("data-state", "hide");
```
Comment (PhilR8) [inline regulations/static/regulations/js/go-to-version.js:25]: don't reload page if hash in close button link matches hash in document URL

File: regulations/static/regulations/js/go-to-version.js [hunk 5]
```diff
@@ -5,25 +5,57 @@ export function goToVersion() {
...
+        }
+    });
+
+    // append current hash to end of closeBtn a href
+    // on load and on hashchange
+    window.addEventListener("pageshow", () => {
+        closeBtn.href = closeBtn.href.split("#")[0] + location.hash;
+    });
+    window.addEventListener("hashchange", () => {
+        closeBtn.href = closeBtn.href.split("#")[0] + location.hash;
```
Comment (PhilR8) [inline regulations/static/regulations/js/go-to-version.js:35]: make sure close button has correct hash on page load and when hash changes

File: regulations/static/regulations/js/go-to-version.js [hunk 6]
```diff
@@ -5,25 +5,57 @@ export function goToVersion() {
...
+        );
         for (const el of view_elements) {
-          el.setAttribute("data-state", "show");
+            el.setAttribute("data-state", "show");
         }
+
+        // add class to content container for scroll-margin-top
+        // when go to version bar is visible
+        const contentContainer = document.querySelector(".content");
+        contentContainer.classList.add("go-to-version");
```
Comment (PhilR8) [inline regulations/static/regulations/js/go-to-version.js:53]: when goToVersion bar is visible, add class to page container for scroll-margin-top when clicking anchor links

File: regulations/templates/regulations/partials/view-and-compare.html [hunk 7]
```diff
@@ -1,10 +1,14 @@
 {% load version_date %}
-<div class="view-and-compare" data-state="hide" data-state-name="view">
+<div id="view-and-compare" class="view-and-compare" data-state="hide" data-state-name="view">
     <div class="view-container">
         <!-- <span class="view-title">View Versions</span> -->
-        View Regulations Effective on
+        <label for="view-options">View Regulations Effective on</label>
```
Comment (PhilR8) [inline regulations/templates/regulations/partials/view-and-compare.html:5]: a11y label

**Review by cgodwin1 — APPROVED** [review]

**Pull request merged**